packages/react/src/__tests__/ReactProfiler-test.internal.js

```diff
@@ -75,7 +67,37 @@ function loadModules({
   };
 }
 
+function loadModulesNoop({
+  enableProfilerTimer = true,
+  enableProfilerCommitHooks = true,
+  enableProfilerNestedUpdatePhase = true,
+  enableProfilerNestedUpdateScheduledHook = false,
+  replayFailedUnitOfWorkWithInvokeGuardedCallback = false,
+} = {}) {
+  loadModules(arguments[0]);
+  ReactNoop = require('react-noop-renderer');
+  ReactDOMClient = null;
+  ReactDOM = null;
+}
+
+function loadModulesReactDOM({
+  enableProfilerTimer = true,
+  enableProfilerCommitHooks = true,
+  enableProfilerNestedUpdatePhase = true,
+  enableProfilerNestedUpdateScheduledHook = false,
+  replayFailedUnitOfWorkWithInvokeGuardedCallback = false,
+} = {}) {
+  loadModules(arguments[0]);
+  ReactNoop = null;
+  ReactDOMClient = require('react-dom/client');
+  ReactDOM = require('react-dom');
+}
```

Comment:
These tests are brittle and depend on import order and test order. Splitting this out to make things a bit more specific as ReactNoop and ReactDOM both being available breaks some tests